Add documentation fro buildpack build time environment variables (#1506)
wso2/docs-choreo-dev: en/docs/develop-components/deploy-an-application-with-buildpacks.md

    e.g.,

    `web: java -jar target/sample.war` <br>

> ### Build Time Environment Variables

Reviewer: ```suggestion
## Configure build-time environment variables
```

Author: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/efb3937a9eac353fbde7d4f1a1c7f007848704f3

> You can configure the environment variables that are necessary to build the component using the Build Configurations Editor in the component Build page. This option is not available for Ballerina and WSO2 MI.

Reviewer: ```suggestion
You can configure the environment variables necessary to build the component using the **Build Configurations Editor** on the component **Build** page. 

!!! info
    The capability to configure build-time environment variables is not available for components created using **Ballerina** or **WSO2 MI** buildpacks.
```

Author: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/efb3937a9eac353fbde7d4f1a1c7f007848704f3

> You can configure the environment variables that are necessary to build the component using the Build Configurations Editor in the Build Tab. This is an optional option and not available for Ballerina and WSO2 MI.

Reviewer 2: ```suggestion
You can configure the environment variables that are necessary to build the component using the Build Configurations Editor in the component Build page. This option is not available for Ballerina and WSO2 MI.
```

Author: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/276829871a1bc1bc09226a8d614bf01e15e70d83

> These environment variables and values will be passed to buildpack during the build time. Hence, this option can be used to configure buildpack specific environment variables or enviromemt variables required for component build.

Reviewer: During the build process, the build-time environment variables and their values are passed to the buildpack. Therefore, you can configure both buildpack-specific environment variables and those required for the component build.

> These environment variables and values will be passed to buildpack during the build time. Hence, this option can be used to configure buildpack specific environment variables or enviromemt variables required for component build.

Reviewer: ```suggestion
During the build process, the build-time environment variables and their values are passed to the buildpack. Therefore, you can configure both buildpack-specific environment variables and those required for the component build.
```

Author: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/efb3937a9eac353fbde7d4f1a1c7f007848704f3

> For example, you can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value. 

Reviewer: For example, if you want to override the Maven command of the **Java** buildpack, you can use `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value.

> For example, you can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value. 

Reviewer: ```suggestion
For example, if you want to override the Maven command of the **Java** buildpack, you can use `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value. 
```

Author: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/abd0d07a2644090eb9f1d9f06bf24e9d028eb409

> - You can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as environment key and `clean install` as value. Refer the following [documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs) for more usages

Reviewer 2: ```suggestion
- You can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as environment key and `clean install` as value. Refer the [documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs) for more usages
```

Author: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/276829871a1bc1bc09226a8d614bf01e15e70d83

> e.g.,
> 
> - You can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as environment key and `clean install` as value. Refer the following [documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs) for more usages

Reviewer 2: ```suggestion
For example, you can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value. 

Refer to the [documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs) for more usages.
```

I think we can separate the external documentation like this

Author: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/276829871a1bc1bc09226a8d614bf01e15e70d83

> Refer to the [documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs) for more usages.

Reviewer: ```suggestion
For more examples, see [Google Cloud's buildpacks documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs).
```

Author: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/efb3937a9eac353fbde7d4f1a1c7f007848704f3
